genie-core/src/main/java/com/netflix/genie/core/jobs/workflow/impl/ApplicationTask.java

```diff
@@ -155,12 +166,26 @@ public void executeTask(
                         );
                         fts.getFile(configFile, localPath);
                     }
+
+                    final long applicationFinish = System.nanoTime();
+                    this.registry.timer(
+                        applicationTimerId
+                            .withTag(MetricsConstants.TagKeys.APPLICATION_ID, applicationId)
```

Comment:
Do you think it's redundant to do the name and the id here? Are you just trying to provide a way to group by easier? Any worry about cardinality?